### ISSUE TITLE
Multiple Heatpumps

### ISSUE THREAD

**Issue by gehoel** [signalkraft/mypyllant-component#15]
I bumped in to the following problem:

We have 2 heat pumps in our home, both with an separate gateway (vr921)
And if I add 2 integrations the entity’s of both heat pumps are joined and not separated.
So it’s not working as it should.

**Comment by signalkraft** [comment]
That's gonna be interesting, since I only have the one heatpump to test. Would you be able to run the commands here https://github.com/signalkraft/myPyllant/blob/main/README.md#contributing and here https://github.com/signalkraft/myPyllant/blob/main/README.md#contributing-test-data ?

This would give me some (anonymized) test data to work with, at least.

**Comment by gehoel** [comment]
Thank you for the response.
I would love to test it, but at the moment for some reason my accounts have been blocked by Vaillant. (both of them)
For over 3 weeks I'm communicating with them to solve the issue.
As soon as the problem is fix I can test.

What I have tested so far is the the information that is collected by your integration is stored in HA.
At this time you should have one separate myVaillant account for each heat pump in myVaillant app from Vaillant.
So I add your integration twice in HA.
But as soon as I add an second myVaillant integration with my other account that data gets stored in the same place.

**Comment by gehoel** [comment]
Vaillant has finally unblocked my account without tell my it was blocked in the first place.
They also made an change so you can now have 2 gateways in one myVaillant account.
This gives already a better result in HA.
In HA I now see 2 heat-pumps and also 2 hydraulic stations.
But Circuit 0, Domestic Hot Water 0 and Zone 1 are showing up ones.

If I go in Domestic hot water 0 there is only 1 Control.
All sensors and Diagnostics are dubbel. (for heat pump 1 and 2) 
When I change the set Point it wil change only one of the sensors.

**Comment by ml1nk** [comment]
Thanks for you work, even now it is more helpful than the official app :)

I also have two systems linked with separate gateways on the same account. The first should have two Circuits and the second three. 

![image](https://github.com/signalkraft/mypyllant-component/assets/4943440/388a33df-622f-4cbf-a9eb-fe17ca621f82)

It only shows a single WP with unknown values and one Circuit more than expected (Neub Fussb) and merges some in Circuit 1 and Circuit 2.

![image](https://github.com/signalkraft/mypyllant-component/assets/4943440/cafdd278-6165-48c6-8954-a2259a9f541d)

Multiple systems found. Please pick a valid --system, choices are 0, 1
~~~
sudo docker run -v $(pwd)/test_data:/build/src/myPyllant/tests/json -ti ghcr.io/signalkraft/mypyllant:latest python3 -m myPyllant.tests.generate_test_data xxx xxx vaillant --country germany --system 0
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/build/src/myPyllant/tests/generate_test_data.py", line 224, in <module>
    asyncio.run(main(args.user, args.password, args.brand, args.country, args.system))
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/build/src/myPyllant/tests/generate_test_data.py", line 91, in main
    if system_index is not None and 0 <= system_index < len(homes):
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<=' not supported between instances of 'int' and 'str'
~~~

~~~
sudo docker run -v $(pwd)/test_data:/build/src/myPyllant/tests/json -ti ghcr.io/signalkraft/mypyllant:latest python3 -m myPyllant.tests.generate_test_data  xxx xxx vaillant --country germany --system 1
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/build/src/myPyllant/tests/generate_test_data.py", line 224, in <module>
    asyncio.run(main(args.user, args.password, args.brand, args.country, args.system))
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/build/src/myPyllant/tests/generate_test_data.py", line 91, in main
    if system_index is not None and 0 <= system_index < len(homes):
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<=' not supported between instances of 'int' and 'str'
~~~

Debug log when refreshing:
[home-assistant_mypyllant_2023-12-10T00-03-22.689Z.log](https://github.com/signalkraft/mypyllant-component/files/13626048/home-assistant_mypyllant_2023-12-10T00-03-22.689Z.log)

**Comment by andreazame** [comment]
Hi, I think I have a similar problem. I have 2 heat pumps with their gateways connected with 2 different accounts, Vaillant didn't let me connect both heat pumps in one single account. 
When I try to connect both my accounts on your integration the first one shows every device and entity, while the second only shows 1 device and 1 entity, to be exact the heating_energy_efficiency_2 sensor.
I tried switching the two accounts and I got the same result: only the first account that I add shows everything.
How can I solve this problem?
Thank you so much for your work
